src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -230,6 +237,174 @@ public SingleRun(DocumentDatabase database, RavenConfiguration configuration, Sc
+                return jsArg.AsString();
+            }
+            
+            private void FillDoubleArrayFromJsArray(double[] array, ArrayInstance jsArray, string signature)
+            {
+                var i = 0;
+                foreach (var (key, value) in jsArray.GetOwnProperties())
+                {
+                    if (key == "length")
+                        continue;
+                    if (value.Value.IsNumber() == false)
+                        throw new ArgumentException($"{signature}: The values argument must be an array of numbers, but got {value.Value.GetType().Name} key({key}) value({value})");
+                    array[i] = value.Value.AsNumber();
+                    ++i;
+                }
+            }
+            
+            private JsValue AppendTimeSeries(JsValue self, JsValue[] args)
+            {
+                AssertValidDatabaseContext("appendTs");
+                
+                const string signature = "appendTs(doc, timeseries, timestamp, tag, values)";
+                const int requiredArgs = 5;
+                
+                if (args.Length != requiredArgs)
+                    throw new ArgumentException($"{signature}: This method requires {requiredArgs} arguments but was called with {args.Length}");
+                
+                var (id, doc) = GetIdAndDocFromArg(args[0], signature);
+
+                string timeseries = GetStringArg(args[1], signature, "timeseries");
+                var timestamp = GetDateArg(args[2], signature, "timestamp");
+                var tag = GetStringArg(args[3], signature, "tag");
+                
+                if(args[4].IsArray() == false)
+                    throw new ArgumentException($"{signature}: The values should be an array but got {args[4].GetType().Name}");
+                var jsValues = args[4].AsArray();
+                var values = ArrayPool<double>.Shared.Rent((int)jsValues.Length);
+                try
+                {
+                    FillDoubleArrayFromJsArray(values, jsValues, signature);
+
+                    var toAppend = new TimeSeriesStorage.Reader.SingleResult
+                    {
+                        Values = new Memory<double>(values, 0, (int)jsValues.Length),
+                        Tag = _jsonCtx.GetLazyString(tag),
+                        Timestamp = timestamp,
+                        Status = TimeSeriesValuesSegment.Live
+                    };
+
+                    _database.DocumentsStorage.TimeSeriesStorage.AppendTimestamp(
+                        _docsCtx,
+                        id,
+                        CollectionName.GetCollectionName(doc),
+                        timeseries,
+                        new[] { toAppend});
+                }
+                finally                
+                {
+                    ArrayPool<double>.Shared.Return(values);
+                }
+                
+                return Undefined.Instance;
```

Comment:
As I see it this is like `void`. The `TimeSeriesStorage.AppendTimestamp` and `RemoveTimestampRange` functions return `LastDatabaseChangeVector` maybe it will be better?

Reply:
Talk with @karmeli87 and @aviv86 what is valuable to return there. JS methods by default are returning undefined so maybe this is enough what you did.

Reply:
I can't think of anything valuable to return here. 
So returning `void` is fine and this also something we can change in the future without breaking any changes.